### PR TITLE
[csv-stringify] Update csv-stringify option to get object as columns

### PR DESCRIPTION
**Pull request by mazamachi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/adaltas/node-csv-stringify/blob/master/src/index.coffee.md#stringifierprototypestringifyline
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/csv-stringify/index.d.ts [hunk 1]
```diff
@@ -13,7 +14,7 @@ declare namespace stringify {
          * Order matters, read the transformer documentation for additionnal information,
          * columns are auto discovered when the user write object, see the "header" option on how to print columns names on the first line.
          */
-        columns?: string[];
+        columns?: object | string[];
```
Comment (rogierschouten) [inline types/csv-stringify/index.d.ts]: A `string[]` is also an object, so this typing doesn't mean anything more than just putting `object`. 
Would it be possible to make the type `columns?: { [index: string]: string } | string[]`?

**Review by rogierschouten — CHANGES_REQUESTED** [review]
Please fix line comment above

**Comment by mazamachi** [comment]
@rogierschouten Thanks for your review.
I fixed as you said, and the fix not to use `object` enabled to remove version restriction!

**Comment by rogierschouten** [comment]
:+1:

**Pull request merged**